python/lsst/ts/ess/csc/data_client/lightning_data_client.py

```diff
@@ -353,16 +356,21 @@ async def process_ld250_noise_or_status(
             close_alarm_status = sensor_data[3] == 0
             severe_alarm_status = sensor_data[4] == 0
             heading = float(sensor_data[5])
-        await self.topics.tel_lightningStrikeStatus.set_write(
-            sensorName=sensor_name,
-            timestamp=timestamp,
-            closeStrikeRate=close_strike_rate,
-            totalStrikeRate=total_strike_rate,
-            closeAlarmStatus=close_alarm_status,
-            severeAlarmStatus=severe_alarm_status,
-            heading=heading,
-            location=device_configuration.location,
-        )
+
+        topic_kwargs = {
+            "sensorName": sensor_name,
+            "timestamp": timestamp,
+            "closeStrikeRate": close_strike_rate,
+            "totalStrikeRate": total_strike_rate,
+            "closeAlarmStatus": close_alarm_status,
+            "severeAlarmStatus": severe_alarm_status,
+            "heading": heading,
+        }
+        # TODO DM-38363 Remove the line containing "if" as soon as XML 16 has
+        #  been released and leave the line setting "location" in place.
+        if hasattr(self.topics.tel_lightningStrikeStatus.DataType(), "location"):
+            topic_kwargs["location"] = device_configuration.location
```

Comment:
I gently suggest testing this ("does this field exist") ONCE when you first construct the data client, and saving the info in a flag. You can probably test it for just one of the two topics -- in which case just a single flag is needed.